WIP - RPP Normalize  U8 - 3D kernel Optimization

Dineshbabu-Ravichandran opened the pull request:
- Optimized Normalize 3D for U8 bit depth using SIMD instructions.

Dineshbabu-Ravichandran committed:
Merge branch 'develop' into db/normalize_1d_host

Dineshbabu-Ravichandran committed:
Implimented Normalize 1d for f32 variant

Dineshbabu-Ravichandran committed:
Implimented Normalize U8 for 1d tensor

Dineshbabu-Ravichandran committed:
Implimentation of normalize 3d Tensor for U8 variant

Dineshbabu-Ravichandran committed:
Changed the ROI

Dineshbabu-Ravichandran committed:
Minor changes on F32

Dineshbabu-Ravichandran committed:
Golden Input and output added

Dineshbabu-Ravichandran committed:
Optimized normalization for images

Dineshbabu-Ravichandran committed:
Resolved small bug

Dineshbabu-Ravichandran committed:
Adjusted checking range

HazarathKumarM committed:
Optimized mean and Std loading

HazarathKumarM committed:
Test Suite support for normalize U8 is added

HazarathKumarM committed:
Merge branch 'develop' into db/normalize_3d_u8_host

Commit:
Changes added in test suite after merging with develop

Commit:
Minor changes on test suite

Commit:
QA support added for normalize 3d U8

Commit:
Minor changes besed on review Commands

Commit:
Minor changes in test suite

Commit:
Addded support for PLN variant

Dineshbabu-Ravichandran committed:
Solved the planar variant QA test failure

Dineshbabu-Ravichandran committed:
Support for pln1 is added

Dineshbabu-Ravichandran committed:
Minor changes

Dineshbabu-Ravichandran committed:
Disabled the Debug mode

Dineshbabu-Ravichandran committed:
Pln1 variant axisMask check added

Dineshbabu-Ravichandran committed:
Merge branch develop into db/normalize_3d_u8_host

Dineshbabu-Ravichandran committed:
Minor changes to improve performance

Dineshbabu-Ravichandran committed:
Added Normalize 1d support on Audio test suite

Dineshbabu-Ravichandran committed:
Merge branch 'develop' into db/normalize_1d_host

Dineshbabu-Ravichandran committed:
Qa support added for normalize 1d

Dineshbabu-Ravichandran committed:
Removed 1d support on Misc test suite

Dineshbabu-Ravichandran committed:
Height paramstrides updated

Dineshbabu-Ravichandran committed:
Minor changes on mean and std updation

Dineshbabu-Ravichandran committed:
Implimented Aligned Length changes

HazarathKumarM committed:
Merge remote-tracking branch 'abishek_rpp/develop' into db/normalize_3d_u8_host

HazarathKumarM commented on utilities/test_suite/HOST/Tensor_misc_host.cpp:
```diff
@@ -178,7 +200,11 @@ int main(int argc, char **argv)
 
                 startWallTime = omp_get_wtime();
-                rppt_normalize_host(inputF32, srcDescriptorPtrND, outputF32, dstDescriptorPtrND, axisMask, meanTensor, stdDevTensor, computeMeanStddev, scale, shift, roiTensor, handle);
+
```
Why do we have the function for u8 in misc tests? you have already added the same function in the image unit tests

Dineshbabu-Ravichandran replied:
I removed U8 call in misc test suite .

HazarathKumarM commented on utilities/test_suite/HOST/Tensor_host.cpp:
```diff
@@ -1470,18 +1487,69 @@ int main(int argc, char **argv)
+                    
+                    if(inputBitDepth == 0)
+                    {
+                        if(srcDescPtr->c == 3)
```
can't we have a single without elseif ? Please check this

HazarathKumarM commented on src/modules/cpu/kernel/normalize.hpp:
```diff
@@ -158,16 +159,39 @@ void compute_3D_mean(Rpp32f *srcPtr, Rpp32f *meanPtr, Rpp32u *dims, Rpp32u *stri
+    meanPtr[0] *= normFactor;
+}
+
+
```
please remove the extra space here

HazarathKumarM commented on src/modules/cpu/kernel/normalize.hpp:
```diff
@@ -197,6 +221,28 @@ void compute_3D_inv_std_dev(Rpp32f *srcPtr, Rpp32f *meanPtr, Rpp32f *stdDevPtr,
+        srcPtrTemp += stride[2];
+    }
+    rpp_rsqrt_avx(stdDevPtr, (Rpp32s)(1), 0, normFactor, scale);
+    
```
please remove the space here also

HazarathKumarM commented on src/modules/cpu/kernel/normalize.hpp:
```diff
@@ -277,6 +323,359 @@ void normalize_3D_tensor_nontoggle(Rpp32f *srcPtr, RpptGenericDescPtr srcGeneric
 }
 
+// Computes normalize for 3D non toggle variants
+void normalize_3D_tensor_nontoggle(Rpp8u *srcPtr, RpptGenericDescPtr srcGenericDescPtr, Rpp8u *dstPtr, RpptGenericDescPtr dstGenericDescPtr,
```
please check the alignment here

HazarathKumarM reviewed (commented):
@Dineshbabu-Ravichandran Please address the review comments

HazarathKumarM committed:
Changed the static buffer to scratch buffer

HazarathKumarM committed:
Support for U8 is removed

Dineshbabu-Ravichandran committed:
Test suite support or normalize is also added on HIP side

Dineshbabu-Ravichandran committed:
Minorr changes based on Review comments

Dineshbabu-Ravichandran committed:
Changed the static memory allocation to strach memory

Dineshbabu-Ravichandran committed:
Changes on scratch memory allocation

Dineshbabu-Ravichandran committed:
Changes on the scratch buffer allocation

Dineshbabu-Ravichandran committed:
Testsuite support for normalize3d on HIP side

Dineshbabu-Ravichandran committed:
Merge branch 'db/normalize_1d_host' into db/normalize_3d_u8_host

Dineshbabu-Ravichandran committed:
QA failure on the PKD3

r-abishek committed:
RPP Warp Perspective on HOST and HIP (#451)

* Make initial changes for raw CPP version of warp perspective

* Fix calls to compute_warp_perspective_src_loc function

* Update changes to go through nearest neighbours case

* AVX HOST codes for warp perspective initial

* Fixes for accuracy in warp perspective

* More fixes for accuracy in warp perspective

* Update the cide for AVX version of Planar to Planar

* Add bilinear u8 host code for warp perspective

* Make updates to include functions for F32 data type

* Make updates to use cast instead of set and fix issues with raw C implementation

* Add i8 host codes

* Add updates for F16 Bilinear Code

* Update the initial HIP code for warp perspective

* Update fixes for HIP code

* Add Warp Perspective Nearest Neighbors F16 code for PKD3_to_PLN3 and PLN3_to_PLN3

* Add updates for PLN to PLN configuration

* Add updates for PKD3 to PKD3 case

* Rename variables

* Update changes to log images separately for Bilinear and Nearest Neighbors

* fixed bug in raw c code of PKD-PLN variant

* minor bug fix for F16 PLN variants

* minor fixes in HOST test suite

* Update the HIP code for review comments and refactoring of device functions

* Update the comments alignment

* Rename functions and add cases in HOST and HIP runTests.py

* Update indentations for compuatations and rename vectors

* Update documentations and add more reference variables

* Make more formatting changes

* Make further updates by including test cases

* Make updates to use reinterpret cast

* Update reinterpret casts for PLN to PLN configuration u8 and i8 codes

* Make updates to enclose code inside AVX2 flag

* Make further changes to update type casting

* Update the version

* Make updates to add warp perspective image

* Modify comments, update CHANGELOG and update flags

* Update further comments in warp perspective

* Add more comments for warp perspective

* Update based on further review comments

* Update the case number for warp_perspective in common.py

* Address review comments

* Make initial changes for raw CPP version of warp perspective

* Fix calls to compute_warp_perspective_src_loc function

* Update changes to go through nearest neighbours case

* AVX HOST codes for warp perspective initial

* Fixes for accuracy in warp perspective

* More fixes for accuracy in warp perspective

* Update the cide for AVX version of Planar to Planar

* Add bilinear u8 host code for warp perspective

* Make updates to include functions for F32 data type

* Make updates to use cast instead of set and fix issues with raw C implementation

* Add i8 host codes

* Add updates for F16 Bilinear Code

* Update the initial HIP code for warp perspective

* Update fixes for HIP code

* Add Warp Perspective Nearest Neighbors F16 code for PKD3_to_PLN3 and PLN3_to_PLN3

* Add updates for PLN to PLN configuration

* Add updates for PKD3 to PKD3 case

* Rename variables

* Update changes to log images separately for Bilinear and Nearest Neighbors

* fixed bug in raw c code of PKD-PLN variant

* minor bug fix for F16 PLN variants

* minor fixes in HOST test suite

* Update the HIP code for review comments and refactoring of device functions

* Update the comments alignment

* Rename functions and add cases in HOST and HIP runTests.py

* Update indentations for compuatations and rename vectors

* Update documentations and add more reference variables

* Make more formatting changes

* Make further updates by including test cases

* Make updates to use reinterpret cast

* Update reinterpret casts for PLN to PLN configuration u8 and i8 codes

* Make updates to enclose code inside AVX2 flag

* Make further changes to update type casting

* Make updates to add warp perspective image

* Modify comments, update CHANGELOG and update flags

* Update further comments in warp perspective

* Add more comments for warp perspective

* Update based on further review comments

* Update the case number for warp_perspective in common.py

* Address review comments

* Fix conflits with warp perspective

* Update version details

* Merge branch 'ar/opt_warp_perspective' of https://github.com/r-abishek/rpp into opt_warp_perspective_rebased

* Update version to 1.9.10 including warp perspective

* Updates to convert to XYWH from LTRB instead of opposite

* Update CHANGELOG.md

Co-authored-by: spolifroni-amd <Sandra.Polifroni@amd.com>

* Revert changes and convert to ltrb instead of xywh

---------

Co-authored-by: Srihari-mcw <srihari@multicorewareinc.com>
Co-authored-by: sampath1117 <sampath.rachumallu@multicorewareinc.com>
Co-authored-by: Kiriti Gowda <kiritigowda@gmail.com>
Co-authored-by: Rajy Rawther <Rajy.MeeyakhanRawther@amd.com>
Co-authored-by: spolifroni-amd <Sandra.Polifroni@amd.com>

kiritigowda committed:
Package - Updates and bugfixes (#488)

* Package - remove clang from test

* CMakeLists - remove BUILD_WITH_AMD_ADVANCE

* Package - Add OMP dependency

* Find Packages - Updates

* Test Package - Deps

* Test - backend Info

* Tests - Fix HIP Test Add

* Test - Add HIP Path

* Test - Find HIP Updates

* Tests - Fix HIP compilation

kiritigowda committed:
Jenkins - Fix Test (#489)

* Jenkins - Fix Test

* Test - Create sepreate test folder

* FileSystem - Find and process

* Find Filesystem - Updates

* Test - Compiler Updates

dependabot[bot] committed:
Docs - Bump rocm-docs-core[api_reference] from 1.11.0 to 1.12.0 in /docs/sphinx (#491)

Bumps [rocm-docs-core[api_reference]](https://github.com/ROCm/rocm-docs-core) from 1.11.0 to 1.12.0.
- [Release notes](https://github.com/ROCm/rocm-docs-core/releases)
- [Changelog](https://github.com/ROCm/rocm-docs-core/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/ROCm/rocm-docs-core/compare/v1.11.0...v1.12.0)

---
updated-dependencies:
- dependency-name: rocm-docs-core[api_reference]
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
Solved QA test failure of Axis Mask 3

Commit:
Merge branch 'develop' into normalize_3d_u8_host

Dineshbabu-Ravichandran committed:
Bug fix for merge with develop
